packages/calcite-components/src/tests/commonTests/cancelableResource.ts

```diff
@@ -0,0 +1,35 @@
+import { mount } from "@arcgis/lumina-compiler/testing";
+import { describe, expect, it, vi } from "vitest";
+import { mockConsole } from "../../tests/utils/logging";
+
+/**
+ * Helper for testing cancelable behavior in components (debounced or throttled).
+ *
+ * @example
+ * describe("cancelable behavior", () => {
+ *   cancelableBehavior("<calcite-action-bar>", "resize"");
+ *   cancelableBehavior("<calcite-color-picker>", "drawColorControls");
+ * });
+ *
+ * @param {ComponentTag} componentTag - The tag name of the component to test.
+ * @param {string} cancelableMethod - The name of the debounced or throttled method to test.
+ */
+export function cancelableBehavior(componentTag: string, cancelableMethod: string): void {
+  describe(`${cancelableMethod} behavior`, () => {
+    mockConsole("warn");
+
+    it(`should cancel pending ${cancelableMethod} on disconnect`, async () => {
+      const componentElement = componentTag.replace(/<|>/g, "");
```

Comment:
If you type as `ComponentTag` (like we do for other utils), you won't need do any string replacement.